source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "rex_engine/types.h"
+#include "rex_std_extra/memory/memory_size.h"
+
+namespace rex
+{
+  struct MemoryStats
+  {
+    // total physical memory
+    rsl::memory_size total_physical_mem{};
+
+    // available physical memory
+    rsl::memory_size avail_physical_mem{};
+
+    // total virtual memory available
+    rsl::memory_size total_virtual_mem{};
+
+    // the page size of the current system, aka how much memory memory is paged to disk
+    rsl::memory_size page_size{};
+
+    // the amount of memory that's currently used by physical memory
+    rsl::memory_size used_physical_mem{};
+
+    // the amount of memory that's paged out to disk
+    rsl::memory_size used_virtual_mem{};
+
+    // amount of page faults that have occurred
+    // Note: this doesn't differentiate between hard and soft page faults
+    // so don't assume that a high page fault count is bad
+    card32 num_page_faults{};
+  };
+} // namespace rex
+
+#ifdef REX_PLATFORM_WINDOWS
+  #include "rex_engine/memory/win/win_mem_stats.h"
```

Comment:
Do we want these conditional compilation statements or do we exclude the file in general from the build?

Reply:
I don't know how you'd exclude this from build other than what I did above, it's a header file, that's never compiled standalone. you can only exclude source files from build

Reply:
I mean exclude it from the project, if it's not present we cannot build it.

Reply:
that's not really true, we use ninja, not vs for building, ninja doesn't have such a thing as projects.
all files that are in the "win" folder are excluded if the target platform is not windows